src/Common/TimerDescriptor.h

```diff
@@ -5,14 +5,24 @@
 namespace DB
 {
 
+enum TimerTypes
+{
+    DEFAULT,
+    RECEIVE_HELLO_TIMEOUT,
+    RECEIVE_TABLES_STATUS_TIMEOUT,
+    RECEIVE_DATA_TIMEOUT,
+    RECEIVE_TIMEOUT,
+};
+
 /// Wrapper over timerfd.
 class TimerDescriptor
 {
 private:
     int timer_fd;
+    int type = TimerTypes::DEFAULT;
```

Comment:
I don't think it is a good idea to store timer type inside `TimerDescriptor`. Cause now it is just used in setter and getter - and is not used by the class itself.